### DISCUSSION TITLE
"-bash: dalatad: command not found" after macos update

### DISCUSSION THREAD

**Discussion by alecristia** [LAAC-LSCP/ChildProject#417] [Show and tell]
the solution is to follow again [the installation instructions](https://childproject.readthedocs.io/en/latest/install.html#macos-users)

**Comment by alecristia** [comment]
oops, that didn't work. I go through the installation but I get "the installation failed: the installer encountered an error that caused the installation to fail. contact the sorftware manufacturer for assistance.". `conda list` yields an error. And if I relaunch the installer, I get "anaconda3 is already installed in XXX. use conda update anaconda to update."

**Reply by alecristia** [reply]
I then tried `rm -rf /Users/acristia/opt/anaconda3/` and went through the installation again, except choosing to install in a specific disk rather than for my user. I got no errors during the installation, but no happy message either, and then conda was not installed. I suspect this may be a connection issue & trying in a different computer next.

**Reply by alecristia** [reply]
I haven't yet tried it, but I wonder if miniconda would suffice, or I really need anaconda?

**Reply by LoannPeurey** [reply]
miniconda should suffice

**Reply by alecristia** [reply]
> I then tried `rm -rf /Users/acristia/opt/anaconda3/` and went through the installation again, except choosing to install in a specific disk rather than for my user. I got no errors during the installation, but no happy message either, and then conda was not installed. I suspect this may be a connection issue & trying in a different computer next.

installation successful in another mac that also had been updated, but still get error for `conda --version`

**Reply by alecristia** [reply]
tried with miniconda, installation successful, but still get this:
> Traceback (most recent call last):
  File "/Users/acristia/opt/miniconda2/bin/conda", line 13, in <module>
    sys.exit(main())
  File "/Users/acristia/opt/miniconda2/lib/python2.7/site-packages/conda/cli/main.py", line 149, in main
    from ..exceptions import conda_exception_handler
  File "/Users/acristia/opt/miniconda2/lib/python2.7/site-packages/conda/exceptions.py", line 19, in <module>
    from ._vendor.auxlib.entity import EntityEncoder
  File "/Users/acristia/opt/miniconda2/lib/python2.7/site-packages/conda/_vendor/auxlib/entity.py", line 253, in <module>
    from .collection import AttrDict, frozendict, make_immutable
  File "/Users/acristia/opt/miniconda2/lib/python2.7/site-packages/conda/_vendor/auxlib/collection.py", line 10, in <module>
    from .compat import isiterable, iteritems, odict, text_type
  File "/Users/acristia/opt/miniconda2/lib/python2.7/site-packages/conda/_vendor/auxlib/compat.py", line 10, in <module>
    from ._vendor.five import WhateverIO as StringIO, with_metaclass
  File "/Users/acristia/opt/miniconda2/lib/python2.7/site-packages/conda/_vendor/auxlib/_vendor/five.py", line 80, in <module>
    absolute_to_nanoseconds = CoreServices.AbsoluteToNanoseconds
  File "/Users/acristia/opt/miniconda2/lib/python2.7/ctypes/__init__.py", line 379, in __getattr__
    func = self.__getitem__(name)
  File "/Users/acristia/opt/miniconda2/lib/python2.7/ctypes/__init__.py", line 384, in __getitem__
    func = self._FuncPtr((name_or_ordinal, self))
AttributeError: dlsym(RTLD_DEFAULT, AbsoluteToNanoseconds): symbol not found


I notice it's talking about python 2.7, perhaps I should kill python 2 first?

**Reply by LoannPeurey** [reply]
I am not sure why it is failing but you should definitely use miniconda3 and not miniconda2 to use python 3.X instead

**Reply by LoannPeurey** [reply]
It does look like you are reinstalling anaconda3 but trying to use anaconda2

**Reply by alecristia** [reply]
thank you for the help! I went at it with an axe:
```
rm -rf PATH-TO-CONDA2
rm -rf PATH-TO-CONDA3
```
then followed instructions to install miniconda, closed my terminal & reopened it, and voilà ! problem fixed.

**Comment by alecristia** [comment]
new problem: now while installing childproject inside the conda env, I get:
> Error: Permission denied @ apply2files - /usr/local/lib/docker/cli-plugins

A search revealed [this issue related to brew](https://github.com/Homebrew/homebrew-core/issues/45009). 

Macos updates break xcode so one should reinstall xcode for brew to work - which I did, but I'm still getting the same error.

Next step is to do `brew update` (done!) then `brew doctor` and handle any errors or warnings it yields. 

However, one of those things to do is `brew cleanup`, but that again gives the same permission denied. I found references to this for docker, but I don't have docker installed (in /Applications). I could try `sudo rm -f /usr/local/lib/docker/cli-plugins` but not sure it's a good idea. @LoannPeurey , thoughts?

**Reply by LoannPeurey** [reply]
not too sure about brew in general, did you try `sudo brew cleanup` ?

**Reply by LoannPeurey** [reply]
alternatively, try `mkdir -p /Applications/Docker.app/Contents/Resources/cli-plugins` then `brew cleanup`

**Reply by alecristia** [reply]
hmm I'll pick up again here tomorrow or next week - thanks for the help!

**Reply by alecristia** [reply]
"tomorrow" took two weeks, but the solution was to do `sudo rm -rf` any offending folder, then the cleanup ran through.

**Comment by alecristia** [comment]
on a different computer, i'm also trying to solve this issue. I could install conda, but 
`conda env create -f env.yml` fails with error:

> Solving environment: failed
ResolvePackageNotFound:   - python=3.7

I tried `conda config --set restore_free_channel true` as per [the top answer here](https://stackoverflow.com/questions/58219956/how-to-fix-resolvepackagenotfound-error-when-creating-conda-environment), but the same error ensued.

Then following [this](https://github.com/conda/conda/issues/7533) tried `conda install python=3.7 `(which is stupid, since this will happen outside the env), so python was installed but of course the error still ensued during env creation.

I'm giving up for now, since I'm still able to do `source ~/ChildProjectVenv/bin/activate` in this machine, but I should sort this out at some point!
